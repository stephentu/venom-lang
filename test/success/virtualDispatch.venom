class A 
  attr value::int
  def self(value::int) = self.value = value; end
  def go() = print(value); end
  def stringify() -> string = "hello"; end
end
a = A(20);
a.go();
print(a);
class B end
print (B());
class C <- A
  def self(value::int) : super(value) = end
  def go() = 
    print ("from C");
    super.go();
  end
end
a = C(30);
a.go();
print(a);
print(a.value);
