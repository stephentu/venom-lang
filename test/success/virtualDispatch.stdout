20
hello
from C
30
hello
30
